### ISSUE TITLE
No such file or directory

### ISSUE THREAD

**Issue by gnzlbg** [rust-lang-nursery/rustfmt#1071]
On linux I get the following error when using cargo fmt after pulling updates from a github repository:

```shell
No such file or directory (os error 2)
usage: cargo fmt [options]
...
```

I don't know how to debug this any further. It would be nice if rustfmt would return a better error when a file is not found (like the name of the file, and where is rustfmt getting that file name from).

**Comment by nrc** [comment]
To clarify - `cargo fmt` used to succeed, you then updated the target source files using Git and then it breaks with the above message? Was this before running `cargo build`? Did running `cargo build`, `cargo update`, or `cargo clean` fix this?

**Comment by gnzlbg** [comment]
> To clarify - cargo fmt used to succeed, you then updated the target source files using Git and then it breaks with the above message? 

Yes. 

> Was this before running cargo build? Did running cargo build, cargo update, or cargo clean fix this?

I only tried cargo test, and that did run fine. 

The way I fixed it was removing the crate directory, and re-cloning it back. I have pulled a couple of times since without issues, if I am able to reproduce it I will leave the directory as is and try to debug it properly. 

Anyhow it might be worth checking all the places in which a file is opened and see if the file name is reported as an error in case it isn't. If the filename is currently always reported I have no idea how I got that error message in the first place.
